src/components/ReportActionItem/MoneyRequestPreview/MoneyRequestPreviewContent.tsx

```diff
@@ -319,6 +313,41 @@ function MoneyRequestPreviewContent({
                                                 </Text>
                                             )}
                                         </View>
+                                        {isScanning && (
+                                            <View style={[styles.flexRow, styles.alignItemsCenter, styles.mt2]}>
+                                                <Icon
+                                                    src={ReceiptScan}
+                                                    height={variables.iconSizeExtraSmall}
+                                                    width={variables.iconSizeExtraSmall}
+                                                    fill={theme.textSupporting}
+                                                />
+                                                <Text style={[styles.textMicroSupporting, styles.ml1, styles.amountSplitPadding]}>{translate('iou.receiptScanInProgress')}</Text>
+                                            </View>
+                                        )}
+                                        {isPending && (
+                                            <View style={[styles.flexRow, styles.alignItemsCenter, styles.mt2]}>
+                                                <Icon
+                                                    src={Expensicons.CreditCardHourglass}
+                                                    height={variables.iconSizeExtraSmall}
+                                                    width={variables.iconSizeExtraSmall}
+                                                    fill={theme.textSupporting}
+                                                />
+                                                <Text style={[styles.textMicroSupporting, styles.ml1, styles.amountSplitPadding]}>{translate('iou.transactionPending')}</Text>
+                                            </View>
+                                        )}
+                                        {!isScanning && hasPendingUI && (
+                                            <View style={[styles.flexRow, styles.alignItemsCenter, styles.mt2]}>
+                                                <Icon
+                                                    src={Expensicons.Hourglass}
+                                                    height={variables.iconSizeExtraSmall}
+                                                    width={variables.iconSizeExtraSmall}
+                                                    fill={theme.textSupporting}
+                                                />
+                                                <Text style={[styles.textLabel, styles.colorMuted, styles.ml1, styles.amountSplitPadding]}>
+                                                    {translate('iou.pendingMatchWithCreditCard')}
+                                                </Text>
+                                            </View>
+                                        )}
```

Comment:
There are some subtle differences in the business logic behind displaying the messages among components. Therefore deciding on whether the message should be displayed should be placed in the components. There is possibility to implement workflow like:

get PENDING_TYPE in the component -> pass PENDING_TYPE to a util function -> return icon and description based on PENDING_TYPE (and nothing more as styling differs among components as well).

However I am concerned it would complicate the code rather than simplify it.

Reply:
Let me know what you think about it.

Reply:
I think that here for example isScanning and isPending can be true at the same time and in that case two messages are being displayed. I agree that it would be more readable but I cannot use this pattern if more than one message should be displayed at the same time.
@yuwenmemon could you please confirm that the situation when more than one message in the preview should be displayed? 
If the answer is no I can apply the pattern.

Reply:
You use a hiererchy here: https://github.com/Expensify/App/pull/40354/files#diff-afdb10cb00c134e119c6ddf52b5bc3a06534d99b7dc325ed625ea9d1234973edR141-R152

And I would say that's what we want to follow. AKA: Only have one of these messages shown at a time, but have a consistent order by which we decide what to show.

The cases where we should have two messages both displayed are rare anyway. It for sure won't happen between having an RTER pending violation and SmartScanning.

Reply:
Great, changed the code to be in line with what is settled